### PR TITLE
clean up code

### PR DESCRIPTION
**Pull request by hannahhoang2704**

### CODE REVIEW

File: eeprom/eeprom.c [hunk 1]
```diff
@@ -3,14 +3,14 @@
...
     i2c_init(i2c0, BAUDRATE);
     gpio_set_function(I2C0_SDA_PIN, GPIO_FUNC_I2C);
     gpio_set_function(I2C0_SCL_PIN, GPIO_FUNC_I2C);
 }
 
-//write an array of data to eeprom
+// write an array of data to eeprom
 void write_to_eeprom(uint16_t memory_address, const uint8_t *data, size_t length) {
     uint8_t buf[ADDR_LEN + length];
     buf[0] = (uint8_t)(memory_address >> 8);    //high byte of memory address
```
Comment (RoniNuutinen) [inline eeprom/eeprom.c]: Minor suggestion: '8' could be named as BYTE.

File: eeprom/eeprom.c [hunk 2]
```diff
@@ -56,12 +56,12 @@ void write_log_entry(const char *str, uint8_t *index) {
         printf("Maximum log entries. Erasing the log to log the messages\n");
         erase_logs(index);
     }
-    size_t size_length = strlen(str); //not include NULL terminator
+    size_t size_length = strlen(str) + 1; //include NULL terminator
```
Comment (RoniNuutinen) [inline eeprom/eeprom.c]: This mysterious +/-1 could be named NULLC, or something of the sort, just as an answer for:
why is 1 added or subtraced here?

File: operation/operation.c [hunk 3]
```diff
@@ -39,15 +37,15 @@ void set_opto_fork_irq() {
                                        &irq_event);
 }
 
-//set piezo interrupt detection with callback event
+// set piezo interrupt detection with callback event
 void set_piezo_irq() {
```
Comment (RoniNuutinen) [inline operation/operation.c:42]: I'm not suggesting this, just offering a thought:
these two functions could be combined as one...?

File: eeprom/eeprom.c [hunk 4]
```diff
@@ -56,12 +56,12 @@ void write_log_entry(const char *str, uint8_t *index) {
...
 
-    if (size_length > STRLEN_EEPROM - 1) {
-        size_length = STRLEN_EEPROM - 1;
+    if (size_length > STRLEN_EEPROM) {
+        size_length = STRLEN_EEPROM;
     }
-    uint8_t log_buf[size_length + 3];
+    uint8_t log_buf[size_length + 2];
```
Comment (RoniNuutinen) [inline eeprom/eeprom.c]: +2 has a macro ADDR_LEN

Comment (RoniNuutinen) [reply eeprom/eeprom.c]: Also.. size_length? I'm assuming string_length

File: operation/operation.c (hunk 3 above)
Comment (hannahhoang2704) [reply operation/operation.c:42]: I would prefer to keep it separate

File: eeprom/eeprom.c (hunk 4 above)
Comment (hannahhoang2704) [reply eeprom/eeprom.c]: that +2 is 2 characters of crc16 so it's nothing related to eeprom ADDR_LEN in macro . Anyway I've changed it

File: eeprom/eeprom.c (hunk 2 above)
Comment (hannahhoang2704) [reply eeprom/eeprom.c]: Done

File: eeprom/eeprom.c (hunk 1 above)
Comment (hannahhoang2704) [reply eeprom/eeprom.c]: Done

**Pull request merged**